### PR TITLE
Config agent spring clean

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

There are a few different changes here. I'd be very happy to split these up into separate PRs if desired. But it should be one change per commit here, as listed below.

All of the changes should be no-op refactors 🤞

1. [**Remove unused methods from ConfigAgent**](https://github.com/guardian/frontend/pull/26007/commits/5e00da93ba599e76716840177ddf4077efdb577d).
    - I couldn't find any uses of these methods, so I've removed them.
2. [**Refactor fetchFrontProperties method**](https://github.com/guardian/frontend/pull/26007/commits/5818220524ce849db06eac9ea0176575ee87f8e1)
    - Mapping over the Option once rather than multiple times in the
  FrontProperties constructor call, and explicitly returning
  FrontProperties.empty if the Option is None, feels more explicit and
  easier to read.
    - Rename from `fetchFrontProperties` to `getFrontProperties` because
  (imo) `fetch` implies that a network call is being made, whereas this
  is actually just looking up a value which should have been fetched
  already by the agent.
3. [**Don't return unneeded data from `refreshAndReturn`**](https://github.com/guardian/frontend/pull/26007/commits/d285b821e722b43bf4553e2327bd96566839b8f9) 
    - The only use of `refreshAndReturn` just maps over the Future it returns
in a for comprehension, it doesn't actually do anything with the value
inside the Future.
    - Using `Box.send` instead of `Box.alter`, and returning `Future[Unit]` makes this more explicit.
    - This is probably the most substantive change...
4. [**Delete unused file**](https://github.com/guardian/frontend/pull/26007/commits/efa248b36cdc997a6eefe57e753e1b43818a8670)
5. [**Add test for `getConfigsUsingCollectionId`**](https://github.com/guardian/frontend/pull/26007/commits/989ef13134e5df3e774f34fcd72ac13ac2d7398f)

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by ioannakok — APPROVED** [review]
This is incredible! Thanks for doing this, great description and very valuable tests 😍

**Comment by ioannakok** [comment]
I don't think there is going to be an issue but it might be a good idea to deploy this to CODE before merging?

**Comment by bryophyta** [comment]
@ioannakok I've tested on CODE and didn't spot any obvious issues. The change that feels most substantive to me is refactoring the return value for `refreshAndReturn`, but I'm fairly confident that the code change should be fine, and I've checked that front press requests from the Fronts tool still work.

If there are any other checks you had in mind that you'd like me to do before merging then just let me know?

File: common/app/services/ConfigAgentTrait.scala [hunk 1]
```diff
@@ -54,12 +54,12 @@ object ConfigAgent extends GuLogging {
     configAgent.alter(Option(config))
   }
 
-  def refreshAndReturn(implicit ec: ExecutionContext): Future[Option[ConfigJson]] =
+  def refreshAndReturn(implicit ec: ExecutionContext): Future[Unit] =
```
Comment (OllysCoding) [inline common/app/services/ConfigAgentTrait.scala:57]: Would this function name still make sense if it's no longer returning anything?

Comment (bryophyta) [reply common/app/services/ConfigAgentTrait.scala:57]: I'll merge as-is but very happy to open a separate PR to rename it if you'd prefer :)

**Pull request merged**